### DISCUSSION TITLE
How to save the output from draw_networkx() to an analyzable format?

### DISCUSSION THREAD

**Discussion by YixiC94** [networkx/networkx#5351] [Q&A]
How do I store NetworkX output NOT as .png or image files, but as an analyzable format with color as node attribute?

Below is the code used to draw NetworkX graph.

```
draw_networkx(
        graph,
        pos=graph.graph['positions'],
        node_color=graph.graph['colors'],
        labels=graph.graph['labels'],
        label=example_presentation,
        font_size=8,
        node_size=800
    )
```

I'd like to save to save the output to a format that we can run various analyses on.

**Comment by rossbar** [comment]
There are many graph formats you can choose from for saving a graph: https://networkx.org/documentation/latest/reference/readwrite/index.html

There is no particular one that is recommended over any other.

Regarding setting the node attributes, that depends on the structure of your graph. From your example, it looks like most of the properties used for visualization are stored on the `.graph` attribute of your graph. If you want to remap them to e.g. the nodes, then you'll have to do so depending on how they are specified. Presumably `graph.graph["positions"]` is a dictionary keyed by node (this is what `draw_networkx` expects), so mapping those positions onto the nodes themselves would look something like:

```python
>>> nx.set_node_attributes(graph, values=graph.graph["positions"], name="position")
```

You'd need to do similar remapping for the other attributes of interest.

Finally, since you brought this up in the context of visualization, you could also use e.g. `nx_agraph` or `nx_pydot` to save the graph in Graphviz's `.dot` format.
